### DISCUSSION TITLE
How to log action created by createAsyncThunk?

### DISCUSSION THREAD

**Discussion by zeroarst** [reduxjs/redux-toolkit#3807] [Q&A] [answered]
I created a middleware like this
```typescript

export const loggerMiddleware: Middleware = () => (next) => (action) => {
  console.log(action.type, { payload: action.payload })
  next(action)
}
```

And my store setup is:
```typescript
export const store = configureStore({
  reducer: {   
    counter: counterReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }).concat(loggerMiddleware)
})
```

I have a thunk created:
```typescript
const fetchSomeData = createAsyncThunk(
  'api/fetchSomeData',
  async (arg) => {
  // call api to get data
  }
)
```

When I call `await dispatch(fetchSomeData(arg))`, I am expecting to see `api/fetchSomeData` in the consle. 
But I cannot see it! I do see `api/fetchSomeData/pending`, `api/fetchSomeData/fullfilled`.

Is logging thunk supported?

**Comment by EskiMojo14** [comment]
are you *dispatching* fetchSomeData? `await dispatch(fetchSomeData(arg))`

**Reply by zeroarst** [reply]
Yes. Sorry the code has been simpified..

**Reply by EskiMojo14** [reply, marked as answer]
okay, you've edited - the behaviour you're seeing is correct.`'api/fetchSomeData'` is a *prefix*, createAsyncThunk creates pending/fulfilled/rejected actions based on it.
also, your middleware has a bug - remember to *return* `next(action)`, otherwise you break other middlewares' ability to return values.
